packages/hardhat/scripts/generateTsAbis.ts

```diff
@@ -1,21 +1,22 @@
 import * as fs from "fs";
 //@ts-expect-error  This script runs after `hardhat deploy --export` therefore its deterministic that it will present
-import allGeneratedContracts from "../../nextjs/generated/hardhat_contracts.json";
+import allGeneratedContracts from "../generated/hardhat_contracts.json";
 import prettier from "prettier";
 
 function main() {
-  const GENERATED_PATH = "../nextjs/generated/hardhat_contracts";
+  const TARGET_DIR = "../nextjs/generated/";
 
   const fileContent = Object.entries(allGeneratedContracts).reduce((content, [chainId, chainConfig]) => {
     return `${content}${parseInt(chainId).toFixed(0)}:${JSON.stringify(chainConfig, null, 2)},`;
   }, "");
 
+  if (!fs.existsSync(TARGET_DIR)) {
+    fs.mkdirSync(TARGET_DIR);
+  }
   fs.writeFileSync(
-    `${GENERATED_PATH}.ts`,
+    `${TARGET_DIR}hardhat_contracts.ts`,
     prettier.format(`export default {${fileContent}} as const;`, { parser: "typescript" }),
   );
```

Comment:
Do we want to keep the `json` file?

It might be confusing: sometimes I search for "hardhat_contract" in my IDE to check that file and now it shows both.

If we remove it, we wouldn't need to create a folder, It could be a tmp file inside the scripts folder that gets removed.

Or am I missing something?

Reply:
Fair point. I figured there was no harm in keeping it and being able to see the actual output, but your point makes sense. I'll remove it. Might even make more sense to just write it in an actual temp dir. I'll fix it :+1: